pkg/rbac/sharedData.go

```diff
@@ -258,15 +258,13 @@ func (cache *Cache) GetDisabledClusters(ctx context.Context) (*map[string]struct
 
 }
 
-func disabledClustersForUser(disabledClusters *map[string]struct{}, userClusters []string) map[string]struct{} {
+func disabledClustersForUser(disabledClusters *map[string]struct{},
+	userClusters map[string]struct{}) map[string]struct{} {
 	userAccessDisabledClusters := map[string]struct{}{}
 	for disabledCluster := range *disabledClusters {
-		for _, cluster := range userClusters {
-			if disabledCluster == cluster { //user has access
-				klog.V(7).Info("user has access to search addon disabled cluster: ", cluster)
-				userAccessDisabledClusters[cluster] = struct{}{}
-
-			}
+		if _, userHasAccess := userClusters[disabledCluster]; userHasAccess { //user has access
+			klog.V(7).Info("user has access to search addon disabled cluster: ", disabledCluster)
```

Comment:
You may want to add the userid in this log.

Reply:
Added